src/main/java/nl/opengeogroep/safetymaps/server/admin/stripes/FotoActionBean.java

```diff
@@ -0,0 +1,154 @@
+    
+    @Override
+    public ActionBeanContext getContext() {
+        return context;
+    }
+
+    @Override
+    public void setContext(ActionBeanContext context) {
+        this.context = context;
+    }
+
+    public List<Map<String, Object>> getFotos() {
+        return fotos;
+    }
+
+    public void setFotos(List<Map<String, Object>> fotos) {
+        this.fotos = fotos;
+    }
+
+    @DefaultHandler
+    public Resolution list() throws NamingException, SQLException {
+        fotos = DB.qr().query("select * from wfs."+TABLE+" order by date", new MapListHandler());
+        return new ForwardResolution(JSP);
+    }
+
+    @DontValidate
+    public Resolution delete() throws Exception {
+        String filename = context.getRequest().getParameter("filename");
+        String location = context.getRequest().getParameter("location");
+        if (filename != null) {
+            try {
+                deleteFromFileSystem(location);
+                DB.qr().update("delete from wfs."+TABLE+" where filename ='" + filename + "'");
+                getContext().getMessages().add(new SimpleMessage("Foto verwijderd."));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        } else {
+            getContext().getMessages().add(new SimpleMessage("Geen bestandsnaam om te verwijderen!"));
+        }
+        list();
+        return new ForwardResolution(JSP);
+    }
+
+    @DontValidate
+    public Resolution downloadFoto() throws Exception {
+        String filename = context.getRequest().getParameter("filename");
+        String location = context.getRequest().getParameter("location");
+        System.out.println("downloading: " + location);
+        File file = new File(location);
+        Path source = Paths.get(location);
+        context.getResponse().setBufferSize(DEFAULT_BUFFER_SIZE);
+        context.getResponse().setContentType(Files.probeContentType(source));
+        context.getResponse().setHeader("Content-Length", String.valueOf(file.length()));
+        context.getResponse().addHeader("content-disposition", "attachment; filename=" + filename);
+        
+        FileInputStream fis = null;
+        BufferedInputStream input = null;
+        BufferedOutputStream output = null;
+
+        try {
+            fis = new FileInputStream(file);
+            input = new BufferedInputStream(fis, DEFAULT_BUFFER_SIZE);
+            output = new BufferedOutputStream(context.getResponse().getOutputStream(), DEFAULT_BUFFER_SIZE);
+
+            
+            byte[] buffer = new byte[DEFAULT_BUFFER_SIZE];
+            int length;
+            while ((length = input.read(buffer)) > 0) {
+                output.write(buffer, 0, length);
+            }
+        } finally {
+            
+            output.close();
+            input.close();
+        }
+        return null;
+    }
+    
+    @DontValidate
+    public byte[] getImage(String fileName) {
```

Comment:
Security vulnerability: user can request any file Tomcat has access to, it is not checked if it is an uploaded photo

Use logging not System.out

Send HTTP 404 when not found or HTTP 500  on errors